File: frontend/oei-web/src/app/domain/model/membership/ethical-charter-signature.ts
export interface EthicalCharterSignature {
  readonly id: string;
  readonly memberId: string;
  readonly version: string;
  readonly signedAt: string;
}

export function createEthicalCharterSignature(fields: EthicalCharterSignature): EthicalCharterSignature {
  return Object.freeze({ ...fields });
}

File: backend/domain/shared/src/main/java/global/oei/domain/shared/charter/EthicalCharterSignature.java
package global.oei.domain.shared.charter;

import java.time.Instant;
import java.util.Objects;
import java.util.UUID;

import global.oei.domain.shared.member.MemberId;

/**
 * Records that a member signed a given version of the OEI ethical charter.
 */
public record EthicalCharterSignature(UUID id, MemberId memberId, String version, Instant signedAt) {

    public EthicalCharterSignature {
        Objects.requireNonNull(id, "id must not be null");
        Objects.requireNonNull(memberId, "memberId must not be null");
        Objects.requireNonNull(version, "version must not be null");
        if (version.isBlank()) {
            throw new IllegalArgumentException("version must not be blank");
        }
        Objects.requireNonNull(signedAt, "signedAt must not be null");
    }
}
